src/modules/person/persistence/person.repo.ts

```diff
@@ -58,4 +57,29 @@ export class PersonRepo {
         await this.em.persistAndFlush(person);
         return this.mapper.map(person, PersonEntity, PersonDo);
     }
+
+    public async findAll(personDo: PersonDo<false>): Promise<Option<PersonDo<true>>[]> {
+        if (personDo.referrer !== undefined && personDo.isInformationBlocked !== undefined) {
+            const result: Option<PersonDo<true>>[] = await this.em.find(PersonEntity, {
```

Comment:
`em.find` should return `PersonEntity[]`. No Option and no DO. You won't need the filter in line 80 anymore.